epaint/src/tessellator.rs

```diff
@@ -242,6 +232,35 @@ pub mod path {
             path.push(center + radius * Vec2::angled(angle));
         }
     }
+
+    // Ensures the radius of each corner is within a valid range
+    fn clamp_radius(corner_radius: Rounding, rect: Rect) -> Rounding {
+        let half_width = rect.width() * 0.5;
+        let half_height = rect.height() * 0.5;
```

Comment:
```suggestion
        let half_height = rect.height() * 0.5;
        let max_cr = half_width.min(half_height);
```

then you only need one call to `.at_most(max_cr)` for each of the corners below (-> faster code)